### DISCUSSION TITLE
How to provide a running instance of POSTGRES to a test under PkgEval on JuliaHub?

### DISCUSSION THREAD

**Discussion by michaelfliegner** [GenieFramework/Genie.jl#494] [Q&A] [answered]
My package BitemporalPostgres has a red PkgEval badge, which I do not like ...
Is it possible or even worth while to try? How to fire up a docker POSTGRES image from inside runtests.jl? Or am I completely off track here?

Please help
Regards Michael

**Comment by AbhimanyuAryan** [comment]
"has a red PkgEval badge" can you please explain this?

**Comment by essenciary** [comment]
@AbhimanyuAryan It's this https://github.com/JuliaCI/PkgEval.jl

@michaelfliegner In general, it might make sense not to test the actual database integration in your unit tests (and do it in separate integration tests). Depending how your tests are set up, you can mock the DB calls. If you don't want to mock, you could use SQLite for tests. Put Genie into `test` env when running the tests and set up the DB connection to SQLite in `test` and Postgres in `dev` and `prod` environments.

**Reply by michaelfliegner** [reply]
Thank You for your time. I do need POSTGRES as sqlite has no intervals. On github CI runs with POSTGRES. So I can live with that. Just that I do not bother about that red PkgEval badge any longer.

**Comment by michaelfliegner** [comment, marked as answer]
I am building an app for audit proof temporal storage ( like for insurance
contracts) and I need POSTGRES for time intervals.

Adrian Salceanu ***@***.***> schrieb am Mi., 9. März 2022,
10:34:
